allennlp/data/dataset_readers/snli.py

```diff
@@ -58,9 +68,14 @@ def read(self, file_path: str):
                 label_field = LabelField(label)
 
                 premise = example["sentence1"]
-                premise_field = TextField(self._tokenizer.tokenize(premise), self._token_indexers)
+                premise_tokens = self._tokenizer.tokenize(premise)
                 hypothesis = example["sentence2"]
-                hypothesis_field = TextField(self._tokenizer.tokenize(hypothesis), self._token_indexers)
+                hypothesis_tokens = self._tokenizer.tokenize(hypothesis)
+                if self._append_null:
```

Comment:
You'll also need to add this to `predict_entailment` in the model, I think.

Reply:
Yeah, you're right, thanks.  Or at least we need to put a note somewhere, and we need to update the `Predictor` to check this...

Reply:
I added a note, but I think to update the `Predictor` cleanly, we should just change how this is done.  It's not high priority right now, though, so I'll open an issue, and we can fix it later.